File: withFunctions/findSinus.cpp
#include<iostream>
#include<cmath>
#define pi 3.14159265358
using namespace std;
int fact(int number){
	int factorial = 1;
	for(int i = number; i >= 1; i--){
		factorial *= i;
	}
	return factorial;
}
int main(){
	int power, counter = 15;
	double result;
	cout << "Angle\tDegree\n";
	for(int i = 0; i < 100; i += 10){
		result = 0;
		double x = (float) i * pi / 180;
		for(int j = 0; j < counter; j++){
			power = 2 * j + 1;
			result = result + pow(-1, j) * pow(x, power) / fact(power);
		}
		printf("%d\t%g\n", i, result);
	}
	return 0;
}

File: withFunctions/findSinus.java
package springIntro;

public class findSinus {

	static int fact(int x) {
		int i, f = 1;
		for(i = x; i >= 1; i--) {
			f = f * i;
		}
		return f;
	}
	
	public static void main(String[] args) {
		System.out.printf("Angle\tDegree\n");
		int pow, i, j, n = 15;
		float x, t;
		for(i = 0; i <= 90; i += 10) {
			t = 0;
			x = (float) ((float) i * Math.PI / 180);
			for(j = 0; j <= n - 1; j++) {
				pow = 2 * j + 1;
				t = (float) (t + Math.pow(-1, j) * Math.pow(x, pow) / fact(pow));
			}
			System.out.printf("%d\t%f\n", i, t);
		}
	}

}